mypy/semanal_namedtuple.py

```diff
@@ -166,6 +166,10 @@ def check_namedtuple_classdef(
                 # And docstrings.
                 if isinstance(stmt, ExpressionStmt) and isinstance(stmt.expr, StrExpr):
                     continue
+                # Allow nested classes.
+                if isinstance(stmt, ClassDef):
+                    self.api.analyze_class(stmt)
```

Comment:
Based on exploring debugger I feel like this line is wrong. Main issue is for this case,

```python
class A(NamedTuple):
  class B:
    ...

  x: A
```

B must be analyzed before field x's type is examined. If I drop this line then reveal_type(A.B) gives desired result and class gets analyzed I think [here](https://github.com/python/mypy/blob/4276308be01ea498d946a79554b4a10b1cf13ccb/mypy/semanal.py#L1724), but it introduces an error of `Name "Bar" is not defined  [name-defined]`. I could maybe add scope here, but sorta feels right path is to analyze nested class defs before processing fields.

Reply:
I'm not sure what's the issue here, but you may need to set up a bunch of state/context in the semantic analyzer to allow nested classes to be processed correctly. For named tuples we probably don't set up all the context right now, since it's not needed to process attributes. 

One idea would be to process named tuple class bodies normally (similar to regular classes) in the semantic analyzer even for named tuples, but we'd skip attribute definitions during this pass. The current namedtuple pass would then ignore everything else except attribute definitions. This would be a more complex change, however.